src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,77 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+
+type Props = {
+  todo: Todo;
+  close: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, close }) => {
+  const [user, setUser] = useState<User | null>();
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    getUser(todo.userId)
+      .then(setUser)
+      .catch(() => setIsError(true));
```

Comment:
you should rather set loading to false on finally, no matter if promise was resolve or rejectred